src/viam/examples/modules/complex/gizmo/api.cpp

```diff
@@ -55,8 +55,10 @@ Gizmo::Gizmo(std::string name) : Component(std::move(name)){};
 
 /* Gizmo server methods */
 
-GizmoServer::GizmoServer() : ResourceServer(std::make_shared<ResourceManager>()){};
-GizmoServer::GizmoServer(std::shared_ptr<ResourceManager> manager) : ResourceServer(manager){};
+GizmoServer::GizmoServer(std::shared_ptr<Server> server)
+    : ResourceServer(std::make_shared<ResourceManager>(), server, this){};
+GizmoServer::GizmoServer(std::shared_ptr<ResourceManager> manager, std::shared_ptr<Server> server)
+    : ResourceServer(manager, server, this){};
```

Comment:
As mentioned in the PR description, general pattern here for all `[TypeOfResource]Server` constructors is to invoke the `ResourceServer` constructor, pass along the resource manager, and server, and also pass the `[TypeOfResource]Server` itself with `this`. Once we're in `resource_server_base`, `this` seems to no longer qualify as a gRPC service and cannot be registered via `register_service()`, so I chose to just pass `this` as a `grpc::Service*`.

Reply:
Yeah that's a good idea. So I took a wack at this: it would certainly be a bit clearer, but it also does not work for some reason. gRPC service registration seems to break a bit and it's nontrivial for me to tell why. All I can see is that every test fails because any call to one our resource clients returns an exception (can't even introspect the error message for some reason). Everything _compiles_ fine but 🤷🏻 . If you feel strongly that we should try to adopt the pattern you're suggesting (which I do think is a good one), I could make a followup ticket for it?

Reply:
I'm not sure it is a good idea to make our classes derive from gRPC classes that way. I think composition, as we are doing now, is going to get you further in the long run and better insulate you from changes in the gRPC world.